ignite/contrib/metrics/regression/r2_score.py

```diff
@@ -18,21 +21,34 @@ class R2Score(_BaseRegression):
         - `y` and `y_pred` must be of same shape `(N, )` or `(N, 1)` and of type `float32`.
     """
 
+    def __init__(
+        self, output_transform: Callable = lambda x: x, device: Union[str, torch.device] = torch.device("cpu")
+    ):
+        self._num_examples = None
+        self._sum_of_errors = None
+        self._y_sq_sum = None
+        self._y_sum = None
+        super(R2Score, self).__init__(output_transform, device)
+
+    @reinit__is_reduced
     def reset(self):
-        self._num_examples = 0
-        self._sum_of_errors = 0
-        self._y_sq_sum = 0
-        self._y_sum = 0
+        self._num_examples = torch.tensor(0, device=self._device)
```

Comment:
Still tensor ?

Reply:
@vfdev-5 Sorry I didn't catch... I didn't see any comment that you said you left above.

Reply:
Oh, I missed your comment here. I thought we already fixed the issue with `self._num_examples` that should be a scalar and not tensor. Maybe, it was another PR, but the point is that we can keep number of samples as a number instead of tensor as we do for other metrics